r/aoa_script.R

```diff
@@ -0,0 +1,179 @@
+library(raster)
+
+#Parameters
+job_name <- 'test'
+
+samplePolygons <- read_sf('samplePolygons.geojson', crs = 4326) #sample Polygons (Dezimalgrad)
+samplePolygon_bbox <- st_bbox(samplePolygons, crs = 4326) #(Dezimalgrad)
+
+aoi <- read_sf('aoi.geojson', crs = 4326) #AOI (Dezimalgrad)
+aoi_bbox <- st_bbox(aoi, crs = 4326) #BBox of AOI (Dezimalgrad)
+
+resolution <- 10 #Resolutin of the Output-Image (Meter)
+cloud_cover <- 15 #Threshold for Cloud-Cover in Sentinel-Images
+t0 <- "2020-01-01"
+t1 <- "2020-12-01"
+timeframe <- paste(t0, '/', t1, sep ="")
+assets = c("B01","B02","B03","B04","B05","B06", "B07","B08","B8A","B09","B11","SCL")
+stac = stac("https://earth-search.aws.element84.com/v0")
+
+mapview(c(st_geometry(samplePolygons),st_geometry(aoi))) #plot aoi and sample Polygons
+
+#############Get Image-Data for AOI
+items_aoi <- stac %>%
+  stac_search(collections = "sentinel-s2-l2a-cogs",
+              bbox = c(aoi_bbox[1],aoi_bbox[2],aoi_bbox[3],aoi_bbox[4]), #Anfrage mit AOI-BBox
+              datetime = timeframe,
+              limit = 100) %>%
+  post_request() 
+items_aoi
+
+collection_aoi = stac_image_collection(items_aoi$features, asset_names = assets, 
+                            property_filter = function(x) {x[["eo:cloud_cover"]] < cloud_cover})
+collection_aoi
+
+targetSystem <- toString(items_aoi$features[[1]]$properties$`proj:epsg`) #read EPSG-Code of Sentinel-Images
+targetString <- paste('EPSG:', targetSystem) #transform EPSG-Code to String
+aoi_transformed <- st_transform(aoi, as.numeric(targetSystem)) #transform AOI to Sentinel-Image EPSG
+aoi_bbox_tranformed <- st_bbox(aoi_transformed, crs = as.numeric(targetSystem)) #derive BBox of transformed AOI
+
+cube_view_aoi = cube_view(srs = targetString,  extent = list(t0 = t0, t1 = t1,
+                                                 left = aoi_bbox_tranformed[1], 
+                                                 right = aoi_bbox_tranformed[3],  
+                                                 top = aoi_bbox_tranformed[4], 
+                                                 bottom = aoi_bbox_tranformed[2]),
+                                                 dx = resolution, 
+                                                 dy = resolution, 
+                                                 dt = "P1D", 
+                                                 aggregation = "median", 
+                                                 resampling = "average")
+
+S2.mask = image_mask("SCL", values=c(3,8,9)) #clouds and cloud shadows
+
+gdalcubes_options(threads = 8) #set Threads for raster cube 
+
+classication_image_name <- paste(job_name, '_classication_image_', sep ="") 
+cube_raster_aoi = raster_cube(collection_aoi, cube_view_aoi, mask = S2.mask) %>%
+  select_bands(c("B02","B03","B04", "B08")) %>%
+  reduce_time(c("median(B02)", "median(B03)", "median(B04)", "median(B08)")) %>%
+  #plot(rgb = 3:1, zlim=c(0,1800))
+  write_tif(
+    dir = "~/GitHub/web-aoa/r/images",
```

Comment:
```suggestion
    dir = "images",
```
Hier sollte ein relativer Pfad verwendet werden. Dadurch lässt sich das R-Script auch für andere leichter ausführen.

> Bitte mit `getwd()` arbeiten, wenn ein absoluter Pfad benötigt wird.